.github/workflows/build-and-deploy.yml

```diff
@@ -0,0 +1,130 @@
+name: Jetstream Build and Deploy
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - '[0-9][0-9][0-9][0-9].[0-9]{1,2}.[0-9]+'
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+    environment: GH Actions
+    permissions:
+      contents: read
+      deployments: write
+      id-token: write
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
```

Comment:
I'm not sure how it interacts with the cache action you have below, but there is also a cache option on this:
```
cache: 'pip'
```